modAionImpl/test/org/aion/zero/impl/db/AionContractDetailsTest.java

```diff
@@ -325,7 +323,8 @@ public void testContractStorageSwitch() {
         sharedProps.setProperty(DatabaseFactory.Props.DB_NAME, "storage");
         ByteArrayKeyValueDatabase storagedb = connectAndOpen(sharedProps, LOG);
         JournalPruneDataSource jpd = new JournalPruneDataSource(storagedb, LOG);
-        original.setDataSource(jpd);
+        AionContractDetailsImpl original = new AionContractDetailsImpl(jpd, null);
+        original.detailsInMemoryStorageLimit = memstoragelimit;
```

Comment:
Is it better to use setter or constructor to assign this argument?

Reply:
I removed most of the places where the variable was set since they were not relevant to the test cases. Also, it is now a static variable updated only during unit tests.